community/values/src/main/java/org/neo4j/values/Values.java

```diff
@@ -291,6 +312,12 @@ public static Value of( Object value )
         {
             return NoValue.NO_VALUE;
         }
+        if ( value instanceof Value )
+        {
+            throw new UnsupportedOperationException(
+                    "Converting a Value to a Value using Values.of() is not supported." );
```

Comment:
perhaps a matter of taste and style, but I would just put `Converting a Value to a Value.` in the message, because the fact that it is not supported is given by the exception type, and the fact that it is using `Values.of()` is given by the stacktrace.